server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -75,6 +94,8 @@
  */
 public class QueryPhase implements SearchPhase {
     private static final Logger LOGGER = LogManager.getLogger(QueryPhase.class);
+    public static final boolean SYS_PROP_LONG_SORT_OPTIMIZED =
+        Booleans.parseBoolean(System.getProperty("es.search.long_sort_optimized", "true"));
```

Comment:
@jimczi  Do we need to have any user-faced documentation for this?  or at least documentation for the support team (e.g. in case sort is misbehaving, disable the optimization)?